File: JULY-2025/25-07-2025/p3487.dart
class Solution {
  int maxSum(List<int> nums) {
    Set<int> numbers = nums.toSet();
    bool positive = false;
    int negativeMaxNumber = -100;
    int maxSum = 0;
    for (var number in numbers) {
      if (number.isNegative) {
        if (number > negativeMaxNumber) {
          negativeMaxNumber = number;
        }
      } else {
        positive = true;
        maxSum += number;
      }
    }
    return positive ? maxSum : negativeMaxNumber;
  }
}

File: JULY-2025/25-07-2025/p3487.rs
use itertools::Itertools;

impl Solution {
    pub fn max_sum(nums: Vec<i32>) -> i32 {
        let m = *nums.iter().max().unwrap();
        if m <= 0 {return  m;}
        nums.into_iter()
                        .filter(|x| *x > 0)
                        .counts()
                        .keys()
                        .sum::<i32>()
    }
}